__tests__/generate.test.tsx

```diff
@@ -66,6 +66,78 @@ describe("GeneratePlaylistPage", () => {
     ).toBeInTheDocument()
   })
 
+  it('should display the error message when the "Create new playlist" option is selected and the playlist name is not filled', async () => {
+    render(<GeneratePlaylistPage {...staticTranslations} />)
+
+    const firstNavigationStepButton = screen.getByRole("button", {
+      name: /steps.step1.title/i,
+    })
+    expect(firstNavigationStepButton).toBeInTheDocument()
+
+    const firstStepContent = screen.getByRole("tabpanel")
+    expect(
+      within(firstStepContent).getByText(/steps.step1.title/i)
+    ).toBeInTheDocument()
+
+    // Select the new playlist option
+    const createNewPlaylistRadio = screen.getByRole("radio", {
+      name: /steps.step1.form.createNewPlaylist.title/i,
+    })
+    await userEvent.click(createNewPlaylistRadio)
+
+    // Enable private playlist
+    const privatePlaylistSwitch = screen.getByRole("switch", {
+      name: /steps.step1.form.createNewPlaylist.privatePlaylist.title/i,
+    })
+    await userEvent.click(privatePlaylistSwitch)
+
+    // Click next
+    const nextButton = screen.getByRole("button", {
+      name: /steps.navigation.next/i,
+    })
+    await userEvent.click(nextButton)
+
+    // Check the error message is displayed
+    const playlistNameError = await waitFor(() => {
+      return screen.getByText(/steps.errors.name.required/i)
+    })
+
+    expect(playlistNameError).toBeInTheDocument()
+  })
+
+  it('should display the error message when the "Use existing playlist" option is selected and the playlist is not selected', async () => {
+    render(<GeneratePlaylistPage {...staticTranslations} />)
+
+    const firstNavigationStepButton = screen.getByRole("button", {
+      name: /steps.step1.title/i,
+    })
+    expect(firstNavigationStepButton).toBeInTheDocument()
+
+    const firstStepContent = screen.getByRole("tabpanel")
+    expect(
+      within(firstStepContent).getByText(/steps.step1.title/i)
+    ).toBeInTheDocument()
+
+    // Select the existing playlist option
+    const useExistingPlaylistRadio = screen.getByRole("radio", {
+      name: /steps.step1.form.useExistingPlaylist.title/i,
+    })
+    await userEvent.click(useExistingPlaylistRadio)
+
+    // Click next
```

Comment:
These comments makes me think that we probably want the code being wrapped into a function with a more explanatory name. Specially for those code snippets that are repeated. For example, these 4 lines of code are repeated at least 3 times in this document.